src/main/webapp/components/Documents.js

```diff
@@ -32,14 +38,94 @@ export const Documents = hh(class Documents extends Component {
   constructor(props) {
     super(props);
     this.state = {
+      alertMessage: '',
+      showAlert: false,
       showAddKeyDocuments: false,
       showAddAdditionalDocuments: false,
       showRemoveDocuments: false,
-      documentToRemove: null
+      documentToRemove: null,
+      error: false,
+      alertType: 'success',
+      invalidEmail: false,
+      collaboratorEmail: '',
     }
     this.removeDocument = this.removeDocument.bind(this);
+    this.getShareableLink = this.getShareableLink.bind(this);
   }
 
+  getShareableLink = () => {
```

Comment:
Curious about why we're adding all of this functionality to the `Documents` component and really inflating that to handle a lot more than it really should. It seems to be this could be broken down into its own small component for easier encapsulation.